src/bond.ts

```diff
@@ -568,57 +623,264 @@ export class BondContract extends Contract {
+   * @param bonds - An array of bond values.
+   */
   setPeriodsBonds(
     senderAddress: IAddress,
     periods: number[],
     bonds: BigNumber.Value[]
   ) {
-    throw new Error('Not implemented');
+    let combinedArray: TypedValue[] = [];
+    periods.map((period, index) => {
+      combinedArray.push(new U64Value(period));
+      combinedArray.push(new BigUIntValue(bonds[index]));
+    });
+
+    const tx = new Transaction({
+      value: 0,
+      data: new ContractCallPayloadBuilder()
+        .setFunction('setPeriodsBonds')
+        .setArgs(combinedArray)
+        .build(),
+      receiver: this.contract.getAddress(),
+      sender: senderAddress,
+      gasLimit: 20_000_000,
+      chainID: this.chainID
+    });
+    return tx;
   }
 
+  /**
+   * Builds a `removeBonds` transaction to remove the bonds for each of the specified periods.
+   * @param senderAddress The address of the sender.
+   * @param periods An array of periods for which the bonds should be removed.
+   */
   removePeriodsBonds(senderAddress: IAddress, periods: number[]) {
-    throw new Error('Not implemented');
+    const inputPeriods = periods.map((period) => new U64Value(period));
+
+    const removeBondsTx = new Transaction({
+      value: 0,
+      data: new ContractCallPayloadBuilder()
+        .setFunction('removePeriodsBonds')
+        .setArgs(inputPeriods)
+        .build(),
+      receiver: this.contract.getAddress(),
+      sender: senderAddress,
+      gasLimit: 20_000_000,
+      chainID: this.chainID
+    });
+    return removeBondsTx;
   }
 
+  /**
+   * Builds a transaction to set the minimum penalty for a sender address.
+   * @param senderAddress - The address of the sender.
+   * @param penalty - The minimum penalty value.
+   */
   setMinimumPenalty(senderAddress: IAddress, penalty: number) {
-    throw new Error('Not implemented');
+    const tx = new Transaction({
+      value: 0,
+      data: new ContractCallPayloadBuilder()
+        .setFunction('setMinimumPenalty')
+        .addArg(new U64Value(penalty))
+        .build(),
+      receiver: this.contract.getAddress(),
+      sender: senderAddress,
+      gasLimit: 20_000_000,
+      chainID: this.chainID
+    });
+    return tx;
   }
 
+  /**
+   * Builds a transaction to set the maximum penalty for a sender address.
+   * @param senderAddress - The address of the sender.
+   * @param penalty - The maximum penalty value.
+   */
   setMaximumPenalty(senderAddress: IAddress, penalty: number) {
-    throw new Error('Not implemented');
+    const tx = new Transaction({
+      value: 0,
+      data: new ContractCallPayloadBuilder()
+        .setFunction('setMaximumPenalty')
+        .addArg(new U64Value(penalty))
+        .build(),
+      receiver: this.contract.getAddress(),
+      sender: senderAddress,
+      gasLimit: 20_000_000,
+      chainID: this.chainID
+    });
+    return tx;
   }
 
+  /**
+   * Builds a transaction to set the withdraw penalty for a given sender address.
```

Comment:
```suggestion
   * Builds a `setWithdrawPenalty` transaction to set the withdraw penalty 
```

Reply:
Hint: senderAddress is the `sender` of the transaction. It doesn't mean that it does some changes for that (sender)address. 
`senderAddress` is the address that will further sign and send this transaction